fix(cli): version tests fail non-deterministically when a new CLI version is published to PyPI

### Submission checklist

- [x] This is a bug, not a usage question.
- [x] I added a clear and descriptive title.
- [x] I searched existing issues and didn't find this.
- [x] I can reproduce this with the latest released version.
- [x] I included a minimal reproducible example and steps to reproduce.

### Area (Required)

- [ ] deepagents (SDK)
- [x] cli
- [ ] acp
- [ ] evals
- [ ] harbor
- [ ] repl
- [ ] daytona
- [ ] modal
- [ ] quickjs
- [ ] runloop
- [ ] langsmith-sandbox
- [ ] Other / not sure / general

### Related Issues / PRs

_No response_

### Reproduction Steps / Example Code (Python)

```python
# From libs/cli, run the version tests:
cd libs/cli
uv run --group test pytest tests/unit_tests/test_version.py -v

# These two tests fail when a newer deepagents-cli version exists on PyPI:
# FAILED test_version_slash_command_omits_update_hint_when_up_to_date
# FAILED test_version_slash_command_includes_release_age
```

### Error Message and Stack Trace (if applicable)

```shell
AssertionError: assert (True, '0.0.41') == (False, None)

  At index 0 diff: True != False

tests/unit_tests/test_version.py:248: AssertionError
```

### Description

Two unit tests in `test_version.py` fail non-deterministically whenever a new `deepagents-cli` version is published to PyPI.

**Root Cause:**
`DeepAgentsApp` fires a background `_check_for_updates()` worker on startup that makes a real `requests.get()` to `https://pypi.org/pypi/deepagents-cli/json`. The existing `_block_sdk_pypi_fetch` autouse fixture patches `get_sdk_release_time` and `CACHE_FILE`, but does NOT patch `is_update_available`, leaving the background CLI update check free to reach PyPI.

When v0.0.41 landed on PyPI, the worker mutated `app._update_available` from `(False, None)` to `(True, "0.0.41")` mid-test, breaking assertions. This also violates the project's `--disable-socket` unit test policy.

**Expected:** Both tests pass reliably regardless of what version is on PyPI.
**Actual:** Tests fail whenever a newer version ships to PyPI.

**Fix:** Patch `is_update_available` inside the autouse fixture (3-line change). I will submit a PR once assigned.


### Environment / System Info

- OS: macOS
- Python: 3.14.3
- deepagents-cli: 0.0.40 (installed from source)
- deepagents (SDK): latest main branch
- Reproduced by running: uv run --group test pytest tests/unit_tests/test_version.py -v

--- a/libs/cli/tests/unit_tests/test_version.py
+++ b/libs/cli/tests/unit_tests/test_version.py
@@ -20,7 +20,13 @@ if TYPE_CHECKING:
 
 @pytest.fixture(autouse=True)
 def _block_sdk_pypi_fetch(tmp_path: Path) -> Iterator[None]:
-    """Prevent `/version` tests from hitting real PyPI for SDK release age.
+    """Prevent `/version` tests from hitting real PyPI for CLI or SDK release age.
+
+    The `DeepAgentsApp` background `_check_for_updates()` worker calls
+    `is_update_available()` on startup, which makes a live PyPI request.
+    Without blocking this, a newly published CLI version on PyPI mutates
+    `app._update_available` mid-test, breaking assertions that assume the
+    initial `(False, None)` state.
 
     Tests that exercise SDK release-age behavior directly override
     `CACHE_FILE` themselves; this fixture only ensures tests that don't care
@@ -30,6 +36,10 @@ def _block_sdk_pypi_fetch(tmp_path: Path) -> Iterator[None]:
     with (
         patch("deepagents_cli.update_check.CACHE_FILE", cache_path),
         patch("deepagents_cli.update_check.get_sdk_release_time", return_value=None),
+        patch(
+            "deepagents_cli.update_check.is_update_available",
+            return_value=(False, None),
+        ),
     ):
         yield
 

